packages/runtime/runtime-utils/src/summarizerNodeContract.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { ISnapshotTree } from "@fluidframework/protocol-definitions";
+import {
+    IGraphNode,
+    ISummarizeInternalResult,
+    ISummarizerNode,
+    ISummarizerNodeConfig,
+    ISummarizerNodeWithGC,
+} from "@fluidframework/runtime-definitions";
+import { SummarizerNode } from "./summarizerNode";
+import { ReadAndParseBlob } from "./summarizerNodeUtils";
+import { SummarizerNodeWithGC } from "./summarizerNodeWithGc";
+
+export { ReadAndParseBlob } from "./summarizerNodeUtils";
+
+export interface ISummarizerNodeRootContract {
+    startSummary(referenceSequenceNumber: number, summaryLogger: ITelemetryLogger): void;
+    completeSummary(proposalHandle: string): void;
+    clearSummary(): void;
+    refreshLatestSummary(
+        proposalHandle: string | undefined,
+        getSnapshot: () => Promise<ISnapshotTree>,
+        readAndParseBlob: ReadAndParseBlob,
+        correlatedSummaryLogger: ITelemetryLogger,
+    ): Promise<void>;
+}
+
+export interface IRootSummarizerNode extends ISummarizerNode, ISummarizerNodeRootContract {}
+
+export interface IRootSummarizerNodeWithGC extends ISummarizerNodeWithGC, ISummarizerNodeRootContract {}
+
+export const createRootSummarizerNode = (
```

Comment:
This is more of a question - Is there any advantage of having a specialized method to do this, rather than the caller just calling "createRoot" on the class directly?
"createRoot" could instead return IRootSummarizeNode in the class?

Reply:
This is just to avoid exporting SummarizerNode

Reply:
There's no `this` here, can't you just alias the static function?

```
export const createRootSummarizerNode = SummarizerNode.createRoot;
```

I guess that obscures the parameters/return type info?

Reply:
As-is it adds a stack frame at runtime though 🤷‍♂️

Reply:
That would be nice, but it would break the docs build. (I suspected as such, but I tried anyway just to make sure!)
> (ae-forgotten-export) The symbol "SummarizerNode" needs to be exported by the entry point index.d.ts

Is the build error I get when I change it to:
```ts
export const createRootSummarizerNode = SummarizerNode.createRoot;
```

Reply:
Thinking about this some more, it might be possible to avoid the doc error and the extra stack frame by going with one of the other options in the larger comment.. I'll take a look.

Reply:
Okay I changed to option (2) above, where I moved the common root definitions to utils, and their respective files.

In order to remove the extra call stack frame and satisfy the docs build, I also moved the `createRoot` functions from being static on the SummarizerNode classes, and made them standalone functions. 

The only real drawback to this approach is that the constructors for the classes needed to become public so the function could call them. I was originally hiding them to prevent accidentally creating them in ways which might violate their expected state, but have just added comments indicating to use the other functions instead.

Let me know what you think!

Reply:
Definitely a fan of private constructors in a case like this but seems like a reasonable compromise.